### ISSUE TITLE
FeatureXML and idXML don’t open as new layer anymore in TOPPView

### ISSUE THREAD

**Issue by timosachsenberg** [OpenMS/OpenMS#3424]
Only new window

**Comment by hroest** [comment]
cannot reproduce, I opened a mzML, then a featureXML and during the opening dialogue I clicked "new layer" and it gave me the following:

![screenshot from 2018-05-01 10-17-55](https://user-images.githubusercontent.com/464014/39476150-f73ea966-4d28-11e8-9dc4-bddeebe9e864.png)

**Comment by jpfeuffer** [comment]
Also can't reproduce on macOS Sierra.

**Comment by hroest** [comment]
I can reproduce this on Windows with the newest develop
![screenshot from 2018-05-01 20-03-23](https://user-images.githubusercontent.com/464014/39500095-1bd4c8ee-4d80-11e8-8865-cf0f600a02b2.png)

**Comment by jpfeuffer** [comment]
Still an issue in the newest workshops. @timosachsenberg

**Comment by hendrikweisser** [comment]
I can reproduce this issue on Ubuntu 18.04. It also affects idXML files.

This is a major problem for us!

**Comment by hendrikweisser** [comment]
> It was minor because I think there is a workaround by just dragging the register into the layer box.

I don't understand that. Can you rephrase?

**Comment by jpfeuffer** [comment]
If I am not mistaken I think what people did, was to drag the register "card" (where you can click through your opened files) and drop it into the upper right box of the file where you want it to be added as a layer. I hope that works.

**Comment by hendrikweisser** [comment]
>> workaround by just dragging the register into the layer box.
> I don't understand that. Can you rephrase?

Okay, I got it. You open the featureXML/idXML in a new window, then drag its entry in the "Layers" box onto the main view tab with the mzML.

**Comment by jpfeuffer** [comment]
Ah yes! Thanks. That was it.

**Comment by hendrikweisser** [comment]
@jpfeuffer: So it's the other way around - drag from the "Layers" onto the tab/register.

**Comment by jpfeuffer** [comment]
I think this bug was introduced with other fixes for Qt5. So you both could be right, that it worked for some time with Qt5, too.

**Comment by hendrikweisser** [comment]
Julianus, would you have time to look into this? Because I don't.

**Comment by hendrikweisser** [comment]
> I think this bug was introduced with other fixes for Qt5. So you both could be right, that it worked for some time with Qt5, too.

There still has to be an aspect that is specific to the Qt5 or OS version. Otherwise I wouldn't be able to reproduce this with the old OpenMS version.

**Comment by hroest** [comment]
> There still has to be an aspect that is specific to the Qt5 or OS version. Otherwise I wouldn't be able to reproduce this with the old OpenMS version.

well there was of course an upgrade of Qt from 5.5 to 5.9 between 16.04 and 18.04: 

- https://packages.ubuntu.com/bionic/libqt5gui5
- https://packages.ubuntu.com/xenial/libqt5gui5

**Comment by timosachsenberg** [comment]
The problem seems to be that
https://github.com/OpenMS/OpenMS/blob/c1988426f8085b7a6bbfd692af5c2af80ae8b0aa/src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp#L1374
is still nullptr (= failed to getActiveSpectrumWidget(); )

**Comment by timosachsenberg** [comment]
Interestingly, if I open the featureXML from recent files list, it works.
So it seems that opening the File Open dialog somehow de-actives the currently active spectrum widget
